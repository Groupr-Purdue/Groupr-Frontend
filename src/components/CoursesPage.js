// @flow

import React from 'react';
import { observer } from 'mobx-react';
import {
  Paper,
  List,
  ListItem,
  Divider,
} from 'material-ui';
import courses from '~/store/courses';
import { Row, Col } from 'react-flexbox-grid-aphrodite';
import router from '~/store/router';
import loadingWrapper from '~/util/loadingWrapper';

const handleClick =
  (course: Object): Function =>
    async (): Void => {
      await courses.register(course.id + 1);
      router.push(`/courses/${course.id + 1}`);
    };

const CoursesPage = (): Object =>
  <Row center='xs'>
    <Col xs={12} sm={6} md={6} lg={6}>
      <Paper>
        <List>
          <Row center='xs'><h3> Your Courses: Select a course to join. </h3></Row>
          <Divider style={{ marginBottom: '10px' }} />
          { do {
            if (courses.list.length)
              courses.list.map((course: Object, idx: number): Element =>
                <ListItem
                  key={idx}
                  id={`course_${idx}`}
                  primaryText={course.title}
                  secondaryText={course.name}
                  onClick={handleClick(course)} />);

            else
              <Row center='xs'>
                <h4> No Courses </h4>
              </Row>;
              // <LinearProgress mode='indeterminate'
              //   style={{ marginBottom: 30 }} />;
          } }
        </List>
      </Paper>
    </Col>
  </Row>;

export { CoursesPage as NakedCoursesPage };

export default loadingWrapper(observer(CoursesPage), courses.loading);
