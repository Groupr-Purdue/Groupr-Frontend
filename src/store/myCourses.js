// @flow

import { observable, action } from 'mobx';
import { BACKEND_URL } from '~/config';
import navbar from '~/store/navbar';
import Loading from '~/store/loading';
import screenResponse from '~/util/screenResponse';
import { testUser } from '~/config';
import reThrow from '~/util/reThrow';

class Courses {
  @observable list = [];
  loading: Loading = new Loading({ waitingFor: 'A class list' });

  @action.bound
  startLoading() {
    this.loading.state = 'loading';
    navbar.loading.state = 'loading';
  }

  @action.bound
  succeedLoading() {
    this.loading.state = 'loaded';
    navbar.loading.state = 'loaded';
  }

  @action.bound
  failLoading() {
    this.loading.state = 'failed';
    navbar.loading.state = 'failed';
  }

  @action.bound
  fetch(): Promise {
    this.loading.waitingFor = 'A class';
    this.startLoading();

    return fetch(
      `${BACKEND_URL}/users/${testUser.id}/courses`,
      {
        method: 'GET',
        headers: { Authorization: testUser.token },
      }
    )
    .then(screenResponse)
    .then((ret: Object): Promise => ret.json())
    .then((json: Object): Array<Object> => {
      this.succeedLoading();
      return this.list = json;
    })
    .catch(reThrow(this.failLoading));
  }
}

export default new Courses();
