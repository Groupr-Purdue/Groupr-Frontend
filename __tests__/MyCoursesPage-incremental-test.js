import React from 'react';
import CoursesPage from '~/components/MyCoursesPage';
import myCourses from '~/store/myCourses';
import { render, shallow } from 'enzyme';

test('jest MyCoursesPages incremental snapshot test', () => {
  expect(shallow(<CoursesPage />)).toMatchSnapshot();
});

describe('MyCoursesPage after loading course 1 with test user', () => {
  beforeEach(done => {
    myCourses.fetch().then(done);
  });

  it('should have the correct course 1 name and title.', () => {
    expect(
      render(<CoursesPage />).find('#course_0').text()
    ).toMatchSnapshot();
  });
});
