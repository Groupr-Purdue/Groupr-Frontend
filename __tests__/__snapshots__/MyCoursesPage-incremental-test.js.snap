// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCoursesPage after loading course 1 with test user should have the correct course 1 name and title. 1`] = `""`;

exports[`jest MyCoursesPages incremental snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Grid>
    <Row
        center="xs"
        middle="xs"
        style={
            Object {
                "height": "50%",
              }
        }
    >
        <Pulse
            duration="1s"
            scaleEnd={1}
            scaleStart={0}
            size={100}
        />
    </Row>
</Grid>,
  "nodes": Array [
    <Grid>
      <Row
            center="xs"
            middle="xs"
            style={
                  Object {
                        "height": "50%",
                      }
            }
      >
            <Pulse
                  duration="1s"
                  scaleEnd={1}
                  scaleStart={0}
                  size={100}
            />
      </Row>
</Grid>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Unknown />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance":  {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "render": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Grid>
          <Row
                    center="xs"
                    middle="xs"
                    style={
                              Object {
                                        "height": "50%",
                                      }
                    }
          >
                    <Pulse
                              duration="1s"
                              scaleEnd={1}
                              scaleStart={0}
                              size={100}
                    />
          </Row>
</Grid>,
        "_debugID": 2,
        "_renderedOutput": <Grid>
          <Row
                    center="xs"
                    middle="xs"
                    style={
                              Object {
                                        "height": "50%",
                                      }
                    }
          >
                    <Pulse
                              duration="1s"
                              scaleEnd={1}
                              scaleStart={0}
                              size={100}
                    />
          </Row>
</Grid>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown />,
}
`;
